src/movies.context.tsx

```diff
@@ -25,14 +24,13 @@ const useMoviesContext = () => {
     const {movies, setMovies} = moviesData;
 
     const addMovie = (movie: InnerState) => {
-        debugger
         setMovies(movies => [...movies, movie])
     };
     const deleteMovie = (index: number) => setMovies((movies) => movies.filter((_movie: any, movieIndex: number) => movieIndex !== index));
-    const editMovie = (movie: Movie, editedIndex: number) => {
+    const editMovie = (movie: Movie | InnerState, editedIndex: number) => {
         const nextMovies = [...movies];
         nextMovies[editedIndex] = movie;
-        setMovies(nextMovies);
+        setMovies(() => nextMovies);
     };
     const searchMovies = (searchField: string) => {
```

Comment:
NIT: would recommend name `searchValue` or something like this... we will have formik hometask and i suspect searchField will cause naming collisions for your future variables... or at least confusion in code readability